dotCMS/src/main/java/com/dotcms/rendering/velocity/services/DotResourceCache.java

```diff
@@ -0,0 +1,220 @@
+/**
+ * 
+ */
+package com.dotcms.rendering.velocity.services;
+
+import java.io.File;
+import java.util.HashSet;
+
+import java.util.Set;
+
+import org.apache.velocity.runtime.RuntimeConstants;
+import org.apache.velocity.runtime.RuntimeServices;
+import org.apache.velocity.runtime.resource.Resource;
+import org.apache.velocity.runtime.resource.ResourceCache;
+import org.apache.velocity.runtime.resource.ResourceManager;
+
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.rendering.velocity.util.VelocityUtil;
+import com.dotcms.repackage.com.google.common.collect.ImmutableSet;
+import com.dotmarketing.business.Cachable;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheAdministrator;
+import com.dotmarketing.business.DotCacheException;
+import com.dotmarketing.util.Logger;
+
+import com.liferay.util.StringUtil;
+
+/**
+ * @author Jason Tesser
+ * @author Andres Olarte
+ * @since 1.6.5 The DotResourceCache was created to allow velocities cache to be distributed across
+ *        nodes in a cluster. It also allows the dotCMS to set velocity to always cache and pull
+ *        from cache Our services methods which generate the velocity files will handle the filling
+ *        and removing of The cache. If something is not in cache though the DotResourceLoader will
+ *        be called.
+ */
+public class DotResourceCache implements ResourceCache, Cachable {
+
+
+    private DotCacheAdministrator cache;
+
+    private String primaryGroup = "VelocityCache";
+    private String macroCacheGroup = "VelocityMacroCache";
+    // region's name for the cache
+    private String[] groupNames = {primaryGroup, macroCacheGroup};
+    private static final String MACRO_PREFIX = "MACRO_PREFIX";
+    private final Set<String> ignoreGlobalVM;
+
+
+
+    public DotResourceCache() {
+        cache = CacheLocator.getCacheAdministrator();
+        String files = System.getProperty(RuntimeConstants.VM_LIBRARY);
+        Set<String> holder = new HashSet<>();
+        for (String file : files.split(",")) {
+            if (file != null) {
+                // System.out.println("FILE: " +file);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)